src/hotspot/share/gc/shared/barrierSetNMethod.cpp

```diff
@@ -54,6 +55,7 @@ int BarrierSetNMethod::nmethod_stub_entry_barrier(address* return_address_ptr) {
   MACOS_AARCH64_ONLY(ThreadWXEnable wx(WXWrite, Thread::current()));
 
   address return_address = *return_address_ptr;
+  AARCH64_ONLY(return_address=pauth_strip_pointer(return_address));
```

Comment:
Style nit: spaces around binary operators please. There are a couple of occurrences.